Firestore/Source/API/FIRDocumentChange.mm

```diff
@@ -63,12 +66,11 @@ + (FIRDocumentChangeType)documentChangeTypeForChange:(const DocumentViewChange &
     NSUInteger index = 0;
     NSMutableArray<FIRDocumentChange *> *changes = [NSMutableArray array];
     for (const DocumentViewChange &change : snapshot.documentChanges) {
-      FIRQueryDocumentSnapshot *document = [FIRQueryDocumentSnapshot
-          snapshotWithFirestore:firestore
-                    documentKey:change.document().key
-                       document:change.document()
-                      fromCache:snapshot.isFromCache
-               hasPendingWrites:snapshot.mutatedKeys.contains(change.document().key)];
+      auto apiDocument = absl::make_unique<QueryDocumentSnapshot>(
+          firestore, change.document().key, change.document(), snapshot.isFromCache,
+          snapshot.mutatedKeys.contains(change.document().key));
```

Comment:
You might consider a named parameter for this last one too.
Or you might create a temporary for it: `bool hasPendingWrites = ^;` It's enough of a mouthful to warrant it (IMO)

Both optional.